src/Test/Utilities/Portable/Compilation/TestOperationVisitor.cs

```diff
@@ -0,0 +1,1116 @@
+            var adds = operation.Adds;
+            // operation.EventReference shouldn't be null. The following conditional logic is
+            // a work around for https://github.com/dotnet/roslyn/issues/23810 and should 
+            // be removed once the issue is fixed
+            if (operation.EventReference == null)
+            {
+                Assert.Equal(LanguageNames.VisualBasic, operation.Language);
+                Assert.Same(operation.HandlerValue, operation.Children.Single());
+            }
+            else
+            {
+                AssertEx.Equal(new[] { operation.EventReference, operation.HandlerValue }, operation.Children);
+            }
+        }
+
+        public override void VisitConditionalAccess(IConditionalAccessOperation operation)
+        {
+            Assert.Equal(OperationKind.ConditionalAccess, operation.Kind);
+            AssertEx.Equal(new[] { operation.Operation, operation.WhenNotNull }, operation.Children);
+        }
+
+        public override void VisitConditionalAccessInstance(IConditionalAccessInstanceOperation operation)
+        {
+            Assert.Equal(OperationKind.ConditionalAccessInstance, operation.Kind);
+            Assert.Empty(operation.Children);
+        }
+
+        internal override void VisitPlaceholder(IPlaceholderOperation operation)
+        {
+            Assert.Equal(OperationKind.None, operation.Kind);
+            Assert.Empty(operation.Children);
+        }
+
+        public override void VisitUnaryOperator(IUnaryOperation operation)
+        {
+            Assert.Equal(OperationKind.UnaryOperator, operation.Kind);
+            var operatorMethod = operation.OperatorMethod;
+            var unaryOperationKind = operation.OperatorKind;
+            var isLifted = operation.IsLifted;
+            var isChecked = operation.IsChecked;
+
+            Assert.Same(operation.Operand, operation.Children.Single());
+        }
+
+        public override void VisitBinaryOperator(IBinaryOperation operation)
+        {
+            Assert.Equal(OperationKind.BinaryOperator, operation.Kind);
+            var operatorMethod = operation.OperatorMethod;
+            var binaryOperationKind = operation.OperatorKind;
+            var isLifted = operation.IsLifted;
+            var isChecked = operation.IsChecked;
+            var isCompareText = operation.IsCompareText;
+
+            AssertEx.Equal(new[] { operation.LeftOperand, operation.RightOperand }, operation.Children);
+        }
+
+        public override void VisitConversion(IConversionOperation operation)
+        {
+            Assert.Equal(OperationKind.Conversion, operation.Kind);
+            var operatorMethod = operation.OperatorMethod;
+            var conversion = operation.Conversion;
+            var isChecked = operation.IsChecked;
+            var isTryCast = operation.IsTryCast;
+            switch (operation.Language)
+            {
+                case LanguageNames.CSharp:
```

Comment:
Same comment: We should be testing to make sure that appropriate exceptions are thrown on wrong-language invocations.

Reply:
> Same comment: We should be testing to make sure that appropriate exceptions are thrown on wrong-language invocations.

Please open an issue to add this.

Reply:
This code was ported from TestOperationWalker.

---
In reply to: [157599042](https://github.com/dotnet/roslyn/pull/23840#discussion_r157599042) [](ancestors = 157599042)